### PR TITLE
feat(assistant): Sentry Assistant

### PR DESCRIPTION
**Pull request by denamwangi**
Opened for discussion of the React approach. This introduces a new Store to track the Guides for Dugout and uses GuideAnchors placed within the code to signal what the Guide will reference. None of the backend is in place so the API call is mocked out. To see this in action got to the org dashboard and add '#test' to the end of the url to trigger the guide (<http://dev.getsentry.net:8000/org-slug/?#test>) 

[Video of intended behavior](https://www.dropbox.com/s/nj4htjuwlnr7lmx/dugout_fe_vid.mov?dl=0)

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/static/sentry/app/components/dugoutHelper.jsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import Reflux from 'reflux';
+import classNames from 'classnames';
+import ApiMixin from '../mixins/apiMixin';
+import GuideStore from '../stores/guideStore';
+
+const DugoutHelper = React.createClass({
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/components/dugoutHelper.jsx]: Can you use the `create-react-class` package? React.createClass will be removed in React 16

**Comment by denamwangi** [comment]
Def! I chatted with Billy IRL and that's one of the edits he suggested.
Thanks Lyn

**Comment by ehfeng** [comment]
[Somewhat large commit](https://github.com/getsentry/sentry/pull/6965/commits/7d3a27b3585370bd0a3d207eb04fbc10cc096ba6), want to explain here why I did it.

I wanted to as much data (or state) away from the UI components into the store. Now, the store will trigger anytime a guide or an anchor is added/removed. When we eventually implement incrementing the step, the guide should once again re-trigger.

There's a function `updateApplicableGuides` (as we add anchors or guides) that recalculates which guides are even possible to show. To do this, I also introduced a new field on guides: `required_anchors` to distinguish between necessary and unnecessary ones.

Finally, the API request itself to the `/assistant/` endpoint is still within the `helper.jsx`. Speaking with @lynnagara, she recommended stores remain stores and API calls be kept in UI components. This is counter to our previous discussion, so I think it's worth mentioning here.

**Pull request closed without merging**